pkg/controller/volume/persistentvolume/controller.go

```diff
@@ -517,14 +518,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
 	}
 }
 
-// updateClaimPhase saves new claim phase to API server.
-func (ctrl *PersistentVolumeController) updateClaimPhase(claim *api.PersistentVolumeClaim, phase api.PersistentVolumeClaimPhase) (*api.PersistentVolumeClaim, error) {
-	glog.V(4).Infof("updating PersistentVolumeClaim[%s]: set phase %s", claimToClaimKey(claim), phase)
-	if claim.Status.Phase == phase {
-		// Nothing to do.
-		glog.V(4).Infof("updating PersistentVolumeClaim[%s]: phase %s already set", claimToClaimKey(claim), phase)
-		return claim, nil
-	}
+// updateClaimStatusWithEvent saves new claim.Status to API server.
+func (ctrl *PersistentVolumeController) updateClaimStatus(claim *api.PersistentVolumeClaim, phase api.PersistentVolumeClaimPhase, volume *api.PersistentVolume) (*api.PersistentVolumeClaim, error) {
```

Comment:
As far as I can tell `updateClaimStatus(...)` is called with `volume *api.PersistentVolume` set to `nil` everywhere except by `updateClaimStatusWithEvent(...). And `updateClaimStatusWithEvent(...) is called with `volume *api.PersistentVolume` nil everywhere. So is volume ever not nil?